pkg/newmapper/cronjob.go

```diff
@@ -0,0 +1,97 @@
+package newmapper
+
+import (
+	"strings"
+	"time"
+
+	"github.com/leanix/leanix-k8s-connector/pkg/kubernetes"
+	"github.com/leanix/leanix-k8s-connector/pkg/mapper"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func GetCronJobs(clusterName string, namespaces *corev1.NamespaceList, kubernetesAPI *kubernetes.API) ([]mapper.KubernetesObject, error) {
+	var allCronJobs []mapper.KubernetesObject
+	for _, namespace := range namespaces.Items {
+		cronJobs, err := kubernetesAPI.CronJobs(namespace.Name)
+		if err != nil {
+			return nil, err
+		}
+		mappedCronJobs, err := MapCronJobs(clusterName, cronJobs)
+		if err != nil {
+			return nil, err
+		}
+		allCronJobs = append(allCronJobs, mappedCronJobs...)
+
+	}
+	return allCronJobs, nil
+}
+
+func MapCronJobs(clusterName string, cronJobs *batchv1.CronJobList) ([]mapper.KubernetesObject, error) {
+	var groupedCronJobs []mapper.KubernetesObject
+	for _, cronJob := range cronJobs.Items {
+		cronJob.ClusterName = clusterName
+		cronJobArtifact, err := CronJobSoftwareArtifact(cronJob)
+		mappedCronJob, err := CronJobDataMapping(cronJob)
+		if err != nil {
+			return nil, err
+		}
+		groupedCronJobs = append(groupedCronJobs, *mappedCronJob, *cronJobArtifact)
+	}
+	return groupedCronJobs, nil
+}
+
+func CronJobSoftwareArtifact(CronJob batchv1.CronJob) (*mapper.KubernetesObject, error) {
+	var CronJobData map[string]interface{}
+	CronJobData = make(map[string]interface{})
+	CronJobId := CronJob.Namespace + "_" + CronJob.Name
+	CronJobData["clusterName"] = CronJob.ClusterName
+	CronJobData["name"] = CronJob.Namespace + ":" + CronJob.Name
+	CronJobData["category"] = "Microservice"
```

Comment:
Is there a reason we want to use abstract interface maps instead of strings maps or typed structs? 
Can the values be anything else than strings? Maybe that's something concerning the other kubernetes objects.

Reply:
Here is a screenshot of an example of all the values inside that CronJobData object. 
![image](https://user-images.githubusercontent.com/73819931/165090366-aedfedf7-6271-46fd-b327-6daf3b7e06da.png)
As you can see it contains a few different data types, thats why I used the abstact interace maps. But if you think we can type this in a better way, please let me know. Not that much of a golang expert yet :)

Reply:
Yeah probably not that easy since we have to rely on them using those data structures